### ISSUE TITLE
Feature Request: Add PublicRouteTable to Outputs of App Cloudformation stack

### ISSUE THREAD

**Issue by TillaTheHun0** [aws/copilot-cli#2660]
Hey Copilot team. I have a feature request to add the `PublicRouteTable` resource, created as part of the root app Cloudformation stack, to the `Outputs` of that stack.

My use case is that I am wanting to add an s3 `VpcEndpoint` to my Copilot VPC as an `addon`. Though, in order to accomplish this, I must provide [a list of Route table ids](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcendpoint.html#cfn-ec2-vpcendpoint-routetableids) to the `VpcEndpoint` resource in Cloudformation.

If the `PublicRouteTable` were added as an output of the app stack, I could use `ImportValue` to pull in the resource into an `addon` stack, similar to what we do for `{ 'Fn::ImportValue': !Sub '${App}-${Env}-EnvironmentSecurityGroup' }`

Perhaps there is another way to accomplish this? Another way to accomplish this would be for Copilot to automatically add a `VpcEndpoint` for common services like `s3`, when the VPC is created, but that is a separate feature request that I suspect would require more consideration. Outputting an already created resource would get me what I need. 😄 

**TL;DR**: Please add the `PublicRouteTable` resource to the `Outputs` of the app root Cloudformation stack.

Thanks for the consideration!

**Comment by huanjani** [comment]
@TillaTheHun0, thanks for this suggestion! It is something we had already considered in the past, and now we have confirmation that it would indeed be useful to users!

**Comment by TillaTheHun0** [comment]
Heya. Another use case for this:

In order to set up VPC peering, one has to add a routing rule to the source VPC routing table, pointing the Peer VPC CIDR block to the VPC peering connection. Adding the routing table id to the Copilot environment CF stack would make this possible without have to reference the static routing table id. We could instead import the value from the Copilot CF stack.

I hope that makes sense!

**Comment by gautam-nutalapati** [comment]
Curious design related question for this feature,
Use case I am thinking of:
- SVC is deploying S3 VPCEndpoint. 
- It is deployed in 2 ENVs A and B.
- Env B is importing Env A VPC resources.
- If Env B is not exporting public RT (As that Env stack did not create VPC resources), SVC deployment in ENV B fails because VPCEndpoint add-on will not be able to reference public routing table output in Env B stack.

Question:
If we import a VPC (instead if opting for default VPC/Subnet creation), should the stack export the public routing table of imported VPC? Similar to how user is prompted to select public/private subnet and then they are exported.

**Comment by TillaTheHun0** [comment]
Revisiting this, I was just able to reference the `PublicRouteTableID` Output on the root Copilot App CF Stack from an Copilot Service Addon CF stack. This satisfies both use cases I provided above. Thank you @Lou1415926 for adding!

I'll close out this issue, but feel free to reopen, if I am missing something. Thanks again, loving Copilot!

**Comment by huanjani** [comment]
Related: #3609 (private route table IDs)
